drivers/watchdog/wdt_esp32.c

```diff
@@ -168,12 +168,15 @@ static int wdt_esp32_init(const struct device *dev)
 
 	wdt_hal_init(&data->hal, config->wdt_inst, MWDT_TICK_PRESCALER, true);
 
-	int ret = esp_intr_alloc(config->irq_source,
-				ESP_PRIO_TO_FLAGS(config->irq_priority) |
-				ESP_INT_FLAGS_CHECK(config->irq_flags),
-				(ISR_HANDLER)wdt_esp32_isr,
-				(void *)dev,
-				NULL);
+	int ret = esp_intr_alloc(
+		config->irq_source,
+		ESP_PRIO_TO_FLAGS(config->irq_priority) |
+#if IS_ENABLED(CONFIG_INPUT_ESP32_TOUCH_SENSOR) || IS_ENABLED(CONFIG_COUNTER_RTC_ESP32)
```

Comment:
What if you just set `ESP_INTR_FLAG_SHARED` flag for all those peripherals regardless it is or isn't enabled? I mean, making the interrupt always shared for those 3-4 peripherals.